blender/LilySurfaceScraper/Scrapers/LocalDirectoryScraper.py

```diff
@@ -21,75 +21,121 @@
 # This file is part of LilySurfaceScraper, a Blender add-on to import materials
 # from a single URL
 
-import zipfile
 import os
-from os import path
-import re
-from urllib.parse import urlparse, parse_qs, urlencode, urljoin
 from .AbstractScraper import AbstractScraper
 
+
 class LocalDirectoryScraper(AbstractScraper):
     """
     This scraper does not actually scrap a website, it links textures from a
     local directory, trying to guess the meaning of textures from their names.
     """
     source_name = "Local Directory"
+    scraped_type = {'MATERIAL', "WORLD", "LIGHT"}
     home_url = None
+    show_preview = False
 
     _texture_cache = None
 
     @classmethod
     def canHandleUrl(cls, url):
         """Return true if the URL can be scraped by this scraper."""
-        return path.isdir(url)
+        return os.path.isdir(url) or os.path.isfile(url)
 
-    def fetchVariantList(self, url):
+    def fetchVariantList(self, path):
         """Get a list of available variants.
         The list may be empty, and must be None in case of error."""
-        self._directory = url
-        return [url]
+        # if asked not to check for subfolders then just return the path given
+        if not self.metadata.deep_check:
```

Comment:
Does `deep_copy` really belong to metadata? It's a bit disturbing that metadata are both an input and an output of the fetchVariantList method.

Reply:
The `deep_check` is just a variable that I'm using to track whether or not to check subfolders. 
it is to differentiate between when the asset picker calls for a local file and when a user loads an asset via a path

The metadata is for a singular asset. `deep_check` gets passed into the metadata in the ScrapedData file when it's initializing the scraper.